include/libdnf/module/module_item_container.hpp

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_MODULE_MODULE_ITEM_CONTAINER_HPP
+#define LIBDNF_MODULE_MODULE_ITEM_CONTAINER_HPP
+
+#include "libdnf/base/base_weak.hpp"
+#include "libdnf/common/weak_ptr.hpp"
+#include "libdnf/module/module_item.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace libdnf::module {
+
+
+class ModuleItemContainer;
+
+
+using ModuleItemContainerWeakPtr = libdnf::WeakPtr<ModuleItemContainer, false>;
+
+
+class ModuleItemContainer {
```

Comment:
We will need a `ModuleSack` for storing fake packages representing modules, so that it can be resolved by libsolv. That's different from `ModuleItemContainer`. But I don't yet understand very well how it will work, so maybe I am mistaken, @j-mracek ?

Reply:
I would recommend to keep it right now like it is, because then I can merge it

Reply:
I also don't know whether that will use `PackageSack` or if the libsolv pool could even be part of the private parts of `ModuleItemContainer` or how it will be done. 

However that is our private thing and we can name that whatever we want, but I think the `ModuleItemContainer` is part of public API and it would be nice it it matched with the rest of Sacks.